src/core/utils/hooks/useContainerWidth.tsx

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useMergedRefs } from './useMergedRefs';
+import { useResizeObserver } from './useResizeObserver';
+
+/**
+ * Hook that returns the width of an element in three stages:
+ *  - initialized with 0
+ *  - immediately set to element's initial width as soon as it's mounted
+ *  - update to new width every time it changes (using `useResizeObserver` hook)
+ *
+ * @private
+ * @param watchResizes If false, ResizeObserver will not be connected and only the initial width will be returned
+ * @returns [ref to attach to the element, stateful width of the element]
+ *
+ * @example
+ * const [ref, width] = useContainerWidth();
+ * ... // do something with width
+ * return <div ref={ref}>...</div>;
+ */
+export const useContainerWidth = <T extends HTMLElement>(
+  watchResizes = true,
+) => {
+  const [contentWidth, setContentWidth] = React.useState(0);
+
+  // callback ref to set initial contentWidth only once
+  const isInitialized = React.useRef(false);
+  const ref = React.useCallback((element: T) => {
+    if (!element) {
+      return;
+    }
+    if (!isInitialized.current) {
```

Comment:
Will it trigger rerender with the same value?

Reply:
what do you mean?

Reply:
oh i see. I only added this because I never ever want to use the width from getBoundingClientRect again after the initial mount, because ResizeObserver should take preference.